runtime/lua/vim/treesitter/query.lua

```diff
@@ -90,13 +90,38 @@ local function new_match_metadata()
   return setmetatable({}, match_metatable)
 end
 
+--- The explicitly set queries from |vim.treesitter.query.set_query()|
+local explicit_queries = setmetatable({}, {
+  __index = function(t, k)
+    local lang_queries = {}
+    rawset(t, k, lang_queries)
+
+    return lang_queries
+  end,
+})
+
+--- Sets the runtime query {query_name} for {lang}
+---
+--- This allows users to override any runtime files and/or configuration
+--- set by plugins.
+---@param lang string: The language to use for the query
+---@param query_name string: The name of the query (i.e. "highlights")
+---@param text string: The query text (unparsed).
+function M.set_query(lang, query_name, text)
+  explicit_queries[lang][query_name] = M.parse_query(lang, text)
+end
+
 --- Returns the runtime query {query_name} for {lang}.
 --
 -- @param lang The language to use for the query
 -- @param query_name The name of the query (i.e. "highlights")
 --
 -- @return The corresponding query, parsed.
 function M.get_query(lang, query_name)
+  if explicit_queries[lang][query_name] then
```

Comment:
So this would not run through the inheritance (modeline) logic that the built in queries go through correct?

Reply:
I like it.  Users can resolve queries how ever they want to with this.

Reply:
This is also what I was thinking of when you submitted your hl-overwrite PR. Which is still useful since you can have your dependent-on-parent language overwrites.

Reply:
:+1: yeah, we can probably still do the PR in nvim-treesitter that was my original one.

Reply:
I think it's woerd to not go through the inheritance logic here, as one might still want to inherit things anyway, and also because it is "explicit" in the modeline when you inherit a query.

I am not sure if I am clear but that feels a bit like "non-standard" if we don't use the modeline in the overriden queries.

Reply:
I want it to be literally: I pass in a string, please do not do anything else with it.

That seems the most straightforward interpretation of `set_query`. You can still construct your query using the modelines and what not, by re-using some of the code we already have, but I'd prefer not to do this magic in `set_query`.

We could add a `literal` parameter to it, but I think that would just complicate things more than necessary.

Reply:
I agree with @tjdevries. I think we could expose the processing we do for query files a separate function if we need to and users could call that if they need that processing.

Reply:
Okay I agree on both of what you said., Let's get that merged.